src/OrchardCore.Modules/OrchardCore.Contents/Services/DefaultContentsAdminListFilter.cs

```diff
@@ -78,7 +78,7 @@ public async Task FilterAsync(ContentOptionsViewModel model, IQuery<ContentItem>
                 {
                     if (ctd.GetSettings<ContentTypeSettings>().Listable)
                     {
-                        var authorized = await _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, Permissions.EditContent, new ContentItem { ContentType = ctd.Name });
+                        var authorized = await _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext.User, CommonPermissions.EditContent, new ContentItem { ContentType = ctd.Name });
```

Comment:
I'm still uncomfortable with the change to both the ContentTypeAuthorizationHandler and not using the contentManager.NewAsync() and setting the owner property if it's required.

i.e. being explicit.

Like as is done here https://github.com/OrchardCMS/OrchardCore/blob/5e3f2c37c93265ecf37a361ae3128d3a1afa7bfe/src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs#L316

and many other places.

I can live with the change to `CommonPermissions` (I don't think it matters much either way to be honest)

Reply:
Well Sebastien doesn't like that I change it to CommonPermissions and last week when I explained him why I was making the change in the ContentTypeAuthorizationHandler he said to keep it because he now understand that it's required for making "Own" permissions variations to return proper results. Now, for using the ContentManager or not, it's a question of taste at this point since it doesn't make any big difference.

I can maybe explain again today in the meeting why I made those changes and why I made them and show you what it fixes. I just merged latest dev yesterday on top of my website and needed to apply this PR on top of it so that I can see the publish/edit buttons for users that only have EditOwnContent on a specific content type.

Reply:
Also if you look closely at the code in the ContentTypeAuthorizationHandler it doesn't make sense to always require to pass a Owner to that method since what we would pass is what we already are looking against in this Handler.

``` C#
        private static bool HasOwnership(ClaimsPrincipal user, ContentItem content)
        {
            if (user == null || content == null)
            {
                return false;
            }

            return user.Identity.Name == content.Owner;
        }
```

Here we pass a User in this method which we would get from the _HttpContextAccessor.HttpContext.User.Identity.Name. So we would basically do : `return _HttpContextAccessor.HttpContext.User.Identity.Name == _HttpContextAccessor.HttpContext.User.Identity.Name;` which is simply passing in that HasOwnership method for nothing.

Also, it means that I need to inject the IHttpContextAccessor everywhere I want to do fake content items and assign those the current Owner... which is an extra step done for nothing too.

Reply:
I've re-read your comment and looked at the code here : 

https://github.com/OrchardCMS/OrchardCore/blob/5e3f2c37c93265ecf37a361ae3128d3a1afa7bfe/src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs#L316-L319

It is true that in this case we are evaluating the same values against each others. Which is because we pass a "fake" Owner. But what we need to evaluate here is if the current user has right to create a content item of that content type only.

So we could easily do `if (!await _authorizationService.AuthorizeAsync(User, Permissions.EditContent, new ContentItem { ContentType = id }))`

So, we are assigning a Owner to the ContentItem before evaluating the permission but we could easily use a Fake ContentItem here again to evaluate without using the ContentManager.NewAsync() returned one.

The only reason we are assigning a Owner here on the ContentItem created is because we want to have it stored on the ContentItem for a later Auth eval.

Reply:
Here what I propose is that we assign a Owner by default in the DefaultContentManager.NewAsync() method like @sebastienros suggested. And we remove the extra `contentItem.Owner = User.Identity.Name;` done everywhere we do so. I suggest we also keep the change I made in the ContentTypeAuthorizationHandler and use it with or without creating a ContentItem with the DefaultContentManager.

Here, I want to point on the fact that when we do a NewAsync() we should evaluate if the current user has the right to create this ContentItem before even calling this NewAsync() method logically too.